src/snowflake/snowpark/functions.py

```diff
@@ -3209,6 +3209,54 @@ def array_intersection(array1: ColumnOrName, array2: ColumnOrName) -> Column:
     return builtin("array_intersection")(a1, a2)
 
 
+def array_generate_range(
+    start: ColumnOrName, stop: ColumnOrName, step: ColumnOrName = None
+) -> Column:
```

Comment:
```suggestion
def sequence(
    start: ColumnOrName, stop: ColumnOrName, step: Optional[ColumnOrName] = None
) -> Column:
```
I think we need to add a function that is named `sequence`. Also updated type hints

Reply:
I'm not sure should we align with the name of the function `sequence` in pyspark since the underline function in SF is called `array_generate_range` and it is not exactly the same as `sequence`. The diff are

1. The 2nd argument `stop` is exclusive in `array_generate_range` while inclusive in `sequence`. Our `array_generate_range` is more similar to the Python `range` function. 
2. The default `step` value is always 1 in `array_generate_range` while it can be -1 in `sequence` when `stop` is larger than `stop`.

While `sequence` probably is a shorter name for this functionality though.